### PR TITLE
Generic webhook handler

### LINKED ISSUE #9: Create a generic webhook handler.
Currently, we can only process webhook events from Paystack. We need to transform that handler into a generic handler for every such Paystack.

### PR DESCRIPTION
**Pull request by subomi**
This webhook handler enables you to receive from any provider with 3 verification methods - `HMAC`, `Basic Auth` & `API Key`.

This resolves #9

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		return errors.New("Configuration cannot be empty")
+	}
+
+	s := strings.NewReader(f)
+	if err := json.NewDecoder(s).Decode(&configStore); err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (jirevwe) [inline config.go:104]: should we not support env vars from the get go, so we don't fall into the trap as convoy

File: providers.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+			AppID: c.AppID,
+		}
+
+		if c.VerifierConfig.HmacConfig != nil {
+			p.verifier = &HmacVerifier{c.VerifierConfig.HmacConfig}
+		} else if c.VerifierConfig.BasicAuthConfig != nil {
+			p.verifier = &BasicAuthVerifier{c.VerifierConfig.BasicAuthConfig}
+		} else if c.VerifierConfig.APIKeyConfig != nil {
+			p.verifier = &APIKeyVerifier{c.VerifierConfig.APIKeyConfig}
+		}
```
Comment (jirevwe) [inline providers.go:34]: This is where the provider verification will be done, yeah?

File: config.go (hunk 1 above)
Comment (Subomi) [reply config.go:104]: @jirevwe It'd have made sense if Convoy had a reusable package for this, write-once adapt everywhere.

File: providers.go (hunk 2 above)
Comment (Subomi) [reply providers.go:34]: This is where we create the verification object per provider.

**Pull request merged**